##  Installing the Dependency (newick module)
The program requires the newick module to handle Newick tree data. You can install this module using the following pip command:

    pip install newick


## Data format

The primary inputs for this program are Fasta files and a Newick tree.

Newick files should adhere to the standard Newick format, similar to those generated by FastTree. Fasta files cannot have stop codons remove all stop codons from the amino acid fasta files.

[FastTree on Galaxy](https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://usegalaxy.org/%3Ftool_id%3Dtoolshed.g2.bx.psu.edu/repos/iuc/fasttree/fasttree/2.1.10%2Bgalaxy1&ved=2ahUKEwim94ie3tyJAxVGM9AFHb6jD8kQFnoECAQQAQ&usg=AOvVaw3Ng8qAuGywusGIA3-THs7n)

We recommend using the fast tree in protine mode. Add an aligned amino acid fasta file into the program and press the Runtool button to generate your tree.

![image](modules/PNGS/FastTree.png)

All Fasta files must be aligned to the HXB2 reference and follow HXB2 numbering


All Fasta files should be width formatted, so the entire sequence is on one line.
[Galaxy Fasta width](https://usegalaxy.org/?tool_id=toolshed.g2.bx.psu.edu%2Frepos%2Fdevteam%2Ffasta_formatter%2Fcshl_fasta_formatter%2F1.0.1)


## config file
    config.py
The inputs for all scripts can be controlled from the *config.py*



## Run Pipline Part 1

*run_pipline_part_1.py*

This script runs step 1-4 from the modules folder.

It converts A Newick (.NWK) file into sepreate small groups then converts those groups into two fasta files that can be used to create a slac file.

### Inputs: 

You can configure the inputs in *config.py*

**Clade** The name used to identify all files associated with a particular run.
Example: *B*

**max_group_size** Max size of the groups the Newick tree is split into.
Example: *150*

**min_group_size** min size of the groups the Newick tree is split into.
Example: *80*

### Input files

**input_newick:** A Newick (.NWK) file containing the full phylogenetic tree. 
Example: *B_rooted_2535.NWK*

**input_amion_acid:** A FASTA file containing the amino acid sequences for the tree must be provided. The sequences should be aligned and follow HXB2 numbering.
Example: *strip_CladeB_AminoAcid_HXBC2(2535).fa*

**input_nucleotide:** A FASTA file with the nucleotide sequences for all taxa in the tree. The sequences should be aligned to and follow HXB2 numbering.
Example: *strip_CladeB_Nucleotide_HXBC2(2535).fa*

**Amino_Root_sequence_path:** A FASTA file containing the amino acid consensus sequence for the clade using HXBC2 numbering.
Example: *Con_amino_B_lin.fa*

**nucleotide_Root_sequence_path:** A FASTA file containing the nucleotide consensus sequence for the clade. 
Example: *Con_nuc_B_lin.fa*

### Outputs: 

**Clade_B_amino_fastas:**
This folder holds all the the resulting amino acid fasta files from the Newick tree being split into groups.

**Clade_B_nucleotide_fastas_cleaned**
This folder holds all the the resulting nucleotide fasta files from the Newick tree being split into groups.


## Galaxy-Workflow-Slac

[Galaxy website](https://galaxyproject.org)

First you will need to create a Galaxy account if you do not already have one.

[Galaxy account](https://training.galaxyproject.org/training-material/faqs/galaxy/account_create.html)

Once you are logged into your Galaxy account you will need to switch to the work flow tab.

Click on the **Workflow** tab on the top menu bar of Galaxy.

![image](modules/PNGS/WorkFlowTab.png)

Click on the **Import** button just under and to the right of the top menu bar of Galaxy.

![image](modules/PNGS/Import_workFlow.png)

Click Browse and find **Galaxy-Workflow-Slac.ga** Upload the workflow to Galaxy.

![image](modules/PNGS/Choose_file.png)

Press the **run workflow** button on the bottom right.

![image](modules/PNGS/Slac.png)

For each group.

Add a fasta file from Clade_B_amino_fastas into Amino acid fasta.
Add a fasta file from Clade_B_nucleotide_fastas_cleaned to nucleic acid fasta.

Then Press the **run workflow** button on the top right.

![image](modules/PNGS/RunWorkFlow.png)

Script: *Galaxy-Workflow-Slac.ga*

Download SLAC files: 

Press the floppy disc (save) button on the bottom left to download the slac jason file.

![image](modules/PNGS/DownLoad.png)

This jason file will need to be named with the group name and saved in Clade_B_Slac_data.

Upload this slac jason file to Hyphy Vison.

[Hyphy Vison](http://vision.hyphy.org/SLAC#tree-tab)

Upload the Slac file to the  HyPhy Viewer website and retrieve the JSON and Newick files associated with each SLAC file from the HyPhy Viewer website. 

Export the file as a jason from the HyPhy Vision website.

![image](modules/PNGS/HyPhyVison.png)

Insure that the name of the JSON and Newick files for each group whose names match and add all files to the file Clade_B_Slac_data.
(e.g., 84.json 84.new) 



## Count Independent Mutations 

run the following script: 
*Count_independent_mutation_Tree.py*

### Inputs: 

You can configure the inputs in *config.py*

**Clade** The name used to identify all files associated with a particular run.
Example: *B*

**clade_consensus** The path to FASTA file used as the original consensus sequence from which the mutants diverged.

**Slac_data**   Path to the SLAC data folder containing the JSON and Newick files.
Example: Clade_B_Slac_data

JSON and Newick files for each group should have matching names.
Example: Group_84.nwk and Group_84.json

### Outputs: 

*Clade_B_Count_independent_mutation.txt*


A text file containing a list of all sites, with each group and the frequency of independent mutation events for each amino acid at those sites within each group.




 